File: src/examples/java/org/vertx/java/examples/eventbusbridge/BridgeServer.java
package org.vertx.java.examples.eventbusbridge;

import org.vertx.java.core.Handler;
import org.vertx.java.core.eventbus.SockJSBridge;
import org.vertx.java.core.http.HttpServer;
import org.vertx.java.core.http.HttpServerRequest;
import org.vertx.java.core.json.JsonObject;
import org.vertx.java.core.sockjs.AppConfig;
import org.vertx.java.deploy.Verticle;

import java.util.ArrayList;
import java.util.List;

/**
 * @author <a href="http://tfox.org">Tim Fox</a>
 */
public class BridgeServer extends Verticle {

  public void start() throws Exception {
    HttpServer server = new HttpServer();

    // Also serve the static resources. In real life this would probably be done by a CDN
    server.requestHandler(new Handler<HttpServerRequest>() {
      public void handle(HttpServerRequest req) {
        if (req.path.equals("/")) req.response.sendFile("eventbusbridge/index.html"); // Serve the index.html
        if (req.path.endsWith("vertxbus.js")) req.response.sendFile("eventbusbridge/vertxbus.js"); // Serve the js
      }
    });

    List<JsonObject> permitted = new ArrayList<>();
    permitted.add(new JsonObject()); // Let everything through
    new SockJSBridge(server, new AppConfig().setPrefix("/eventbus"), permitted);

    server.listen(8080);
  }
}

File: vertx-examples/src/main/javascript/eventbusbridgelistener/bridge_server.js
load('vertx.js')

var server = vertx.createHttpServer()
var sockjsServer;

var log = [];

// Serve the static resources
server.requestHandler(function(req) {
  if (req.uri == "/") req.response.sendFile("eventbusbridgelistener/index.html")
  if (req.uri == "/log.json") req.response.end(JSON.stringify(log));
  if (req.uri == "/vertxbus.js") req.response.sendFile("eventbusbridgelistener/vertxbus.js")
});


var myListener = {
  clientDisconnected: function(clientId) {
    log.push(clientId + ' disconnected');
  }
};

// Create a SockJS bridge which lets everything through (be careful!)
vertx.createSockJSServer(server).setEventBusBridgeListener(myListener).bridge({prefix: "/eventbus"}, [{}], [{}]);

server.listen(8080);